### DISCUSSION TITLE
VPN Disconnects After Startup in macOS

### DISCUSSION THREAD

**Discussion by Hermione2408** [mitmproxy/mitmproxy#7425] [General]
I am encountering an issue where my VPN connection disconnects immediately after starting. Below are the details of my setup and the problem:

Application Details:

- Application Name: Mitmproxy Redirector
- macOS Version: macOS 13.5

Problem:
After invoking startVPNTunnel() in the PacketTunnelProvider, the VPN successfully initializes but disconnects immediately with no detailed error message.

Logs:
```
error 18:38:23.440491+0530 Mitmproxy Redirector VPN disconnected after startup
error 18:38:23.440803+0530 Mitmproxy Redirector Proxy setup failed: VPN failed to maintain connection
default	13:07:13.903198+0530	sysextd	/Applications/Mitmproxy Redirector.app/Contents/Library/SystemExtensions/io.demo.macos-redirector.network-extension.systemextension: entitlement `com.apple.developer.endpoint-security.client` not present or not true
```

Questions:

1. Is the com.apple.developer.endpoint-security.client entitlement necessary for this application’s functionality?
2. Are there specific settings or configurations I might have missed to ensure the VPN connection remains active?

Could you please provide guidance on resolving this issue?

**Comment by mhils** [comment]
This looks like a duplicate of https://github.com/mitmproxy/mitmproxy/issues/7419.
